### PR TITLE
Block builder should fail when schema is null

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
Checks if the schema or blinded schema is not null according to the block builder `blinded` property.
An issue was reported on [discord](https://discord.com/channels/697535391594446898/1050616638497640548/1142112922168995900) by @jtraglia:
With the previous implementation, `builder.blindedSchema(validSchema).schema(null).build()` would pass

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by jtraglia — APPROVED** [review]
LGTM 👍 thanks for making this PR!

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/bellatrix/BeaconBlockBodyBuilderBellatrix.java [hunk 1]
```diff
@@ -70,7 +70,11 @@ public BeaconBlockBodyBuilder executionPayloadHeader(
 
   @Override
   protected void validateSchema() {
-    checkState(schema != null || blindedSchema != null, "schema or blindedSchema must be set");
+    if (isBlinded()) {
+      checkState(blindedSchema != null, "blindedSchema must be set");
```
Comment (rolfyone) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/bellatrix/BeaconBlockBodyBuilderBellatrix.java]: to me the 'nit' was we should be doing something like
```
checkState(blindedSchema != null && schema == null, "The blindedSchema must be set, with no schema set.")
```

Comment (rolfyone) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/bellatrix/BeaconBlockBodyBuilderBellatrix.java]: @jtraglia ?

**Review by rolfyone — APPROVED** [review]
marking approved, its a small change if one is required at all...

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/bellatrix/BeaconBlockBodyBuilderBellatrix.java (hunk 1 above)
Comment (mehdi-aouadi) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/bellatrix/BeaconBlockBodyBuilderBellatrix.java]: Makes sense. I updated the checks and the tests accordingly

Comment (jtraglia) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/blockbody/versions/bellatrix/BeaconBlockBodyBuilderBellatrix.java]: Sorry for the slow reply. Yes, this what I originally had in mind. Thanks.

**Pull request merged**